### DISCUSSION TITLE
Ошибка доступа при переходе в цикл если первая таблица недоступна для пользователя

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#42] [Как это сделать?] [answered]
Задача:
В таблице циклов 3 расчетных таблицы. Пользователю нужно дать доступ к самой таблице циклов и только к последней расчетной из них (например, с сортом 30) 
При переходе из таблицы циклов переход осуществляется только в первую с сортом 10, к которой доступ запрещён. 

Вопрос:
Как попадать не в первую, а в любую другую?

Спасибо!

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Нужно модифицировать код кнопки Открыть. Базово она не учитывает доступы к таблицам:

```
= : linkToTable(table: $table; cycle: #id; target: 'self')

table: $tableSearch{var: "t" = $tables}
tableSearch: if(condition: $#t = $#lc; then: $error; else: $tableCheck)
	tableCheck: if(condition: cond`$#t[0][edit_roles] = $#nr || $#t[0][read_roles] = $#nr`; then: $#t[0][id]; else: $tableSearch{var: "t" = $tablesCut})
		tablesCut: listCut(list: $#t; cut: "first"; num: 1)
		
error: errorException(text: "Нет доступных таблиц. Обратитесь к администратору системы.")

tables: selectRowList(table: 'tables'; field: 'edit_roles'; field: 'read_roles'; field: 'id' ; where: 'type' = "calcs"; where: 'tree_node_id' = $#nti; order: 'sort' )
```

Здесь логично использовать рекурсивный перебор, тк его можно прервать как только мы нашли доступную таблицу. В отличии от рекурсии `listReplace` будет идти до конца списка.
